cEP-0020.md

```diff
@@ -0,0 +1,418 @@
+based on points threshold and a newcomer will also get some special privileges
+after the completion of each level. There will also be different badges on the
+basis of performed activities by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to become a developer at coala or to get the
+**coala Developer** badge.
+
+## Background
+
+### Newcomer Experience
+
+**The problems we have:**
+
+- How to attract more newcomers
+  and how to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  and how to use those feedback to improve coala and its processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to open source for the first time, and additional barriers while attempting
+to contribute to projects run by the coala organization.
+
+Some barriers they face include orientation issues that can potentially
+demotivate newcomers from placing their first contribution. On the other
+hand, gamification is widely used to engage and motivate people to
+accomplish tasks and improve their performance.
+
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Introduction
+
+There will be four game design elements to help newcomers: Activities, Points,
+Badges and Levels. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create a bug report
+- Create newcomer difficulty issue
+- Create low difficulty issue
+- Create medium difficulty issue
+- Create a bear proposal issue
+- Create pull request that fixes a bug
+- Create pull request for bear proposal
+- Create pull request for newcomer issue
+- Create pull request for low difficulty issue
+- Create pull request for medium difficulty issue
+- Gitmate errors on pull request
+- CI status of the pull request
+- Review comments on pull request
+- Messages on Gitter channels
+  - Good messages
+  - Bad messages
+- Introduce `.coafile` in other projects
+
+### Points
+
+Points are a simple tool to reward behavior, and they provide an excellent
+base upon which to build a gamified design. Now let's assign some points
+to each of the tracked activities mentioned above:
+
+- Create Issue
+  - Points will only be given when the issue has an extra label `process/verified`
```

Comment:
how are you going to know that a maintainer added label `process/verified` -- does IGitt support that?

fwiw, this cEP will not be accepted while it creates more work for maintainers.

you need to work harder to find better ways of determining whether the issue is good enough to be counted.

Reply:
Normally, if an issue has been open for several days, without labels like `question`, `needs info`, we could consider that as a "verified" issue IMO.

Reply:
The main problem is when a new issue is opened, **gitmate** label it, and it may be wrong and it is wrong in most of the cases.

And I am giving points based on labels like `difficulty/low` issue will earn more number of points then `diffulty/newcomer` issue.

So, with varifying the validity of the issues we also have to varify if the `labels` on the issues are correct or not.

Reply:
Maintainers will update the label if it was inappropriate. Maybe your system needs to update the database accordingly.

Reply:
Why should we give points until we are not sure if it is varified or not? And Is it good to first give someone points then deduce it, for our bot mistakes? That would not be a very good gamification design IMO.
>Maintainers will update the label if it was inappropriate

That will again create more works for maintainers
>fwiw, this cEP will not be accepted while it creates more work for maintainers

@li-boxuan, what do you think about [this]?(https://github.com/coala/cEPs/pull/150#discussion_r189136423)
Let the first developers do their work. And if the levels seems inappropriate to them, they can ask maintainers to label it, And only then they give the thumbs up.

This way a maintainer will not have to look at every new opening issue.